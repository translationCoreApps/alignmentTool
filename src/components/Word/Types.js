export const SECONDARY_WORD = 'secondary';
export const PRIMARY_WORD = 'primary';
