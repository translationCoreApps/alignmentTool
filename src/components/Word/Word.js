import React, { Component } from 'react';
import PropTypes from 'prop-types';
import WordOccurrence from './WordOccurrence';

const internalStyle = {
  borderLeft: '5px solid #44C6FF',
  padding: '5px',
  margin: '10px',
  backgroundColor: '#FFFFFF',
  boxShadow: "0 1px 4px rgba(0, 0, 0, 0.3), 0 0 40px rgba(0, 0, 0, 0.1) inset",
  cursor: 'move',
  display: 'flex',
  flexDirection: 'row'
};

/**
 * Renders a single word
 * @property {string} word - the represented word
 * @property {int} occurrence
 * @property {int} occurrences
 */
class Word extends Component {
  render() {
    const { word, isDragging, occurrence, occurrences } = this.props;
    const opacity = isDragging ? 0.4 : 1;

    return (
      <div style={{ ...internalStyle, opacity }}>
        <span style={{flex: 1}}>
          {word}
        </span>
        <WordOccurrence occurrence={occurrence}
                        occurrences={occurrences}/>
      </div>
    );
  }
}

Word.propTypes = {
  word: PropTypes.string.isRequired,
  occurrence: PropTypes.number.isRequired,
  occurrences: PropTypes.number.isRequired,
  isDragging: PropTypes.bool.isRequired
};

export default Word;
