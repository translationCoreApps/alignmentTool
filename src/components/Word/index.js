import React from 'react';
import PropTypes from 'prop-types';
import Word from './Word';
import { DragSource } from 'react-dnd';
import ItemTypes from '../ItemTypes';

/**
 * Renders a draggable word.
 *
 * @property {string} word - the represented word
 * @property {int} occurrence
 * @property {int} occurrences
 */
class DraggableWord extends React.Component {
  render() {
    const {connectDragSource, word, occurrence, occurrences, isDragging} = this.props;
    return connectDragSource(
      <Word word={word}
            occurrence={occurrence}
            occurrences={occurrences}
            isDragging={isDragging}/>
    );
  }
}

DraggableWord.propTypes = {
  word: PropTypes.string.isRequired,
  occurrence: PropTypes.number.isRequired,
  occurrences: PropTypes.number.isRequired,
  connectDragSource: PropTypes.func.isRequired,
  isDragging: PropTypes.bool.isRequired
};

/**
 * Handles drag events on the word
 */
const dragHandler = {
  beginDrag(props) {
    // Return the data describing the dragged item
    return {
      word: props.word,
      occurrence: props.occurrence,
      occurrences: props.occurrences,
      type: ItemTypes.BOTTOM_WORD
    };
  },
  endDrag() { // receives: props, monitor, component
    // When dropped on a compatible target, do something
    // const item = monitor.getItem();
    // console.log(item)
  }
};

export default DragSource(
  ItemTypes.BOTTOM_WORD,
  dragHandler,
  (connect, monitor) => ({
    connectDragSource: connect.dragSource(),
    isDragging: monitor.isDragging()
  })
)(Word);
