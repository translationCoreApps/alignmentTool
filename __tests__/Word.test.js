import React from 'react';
import renderer from 'react-test-renderer';
import Word from '../src/components/Word/Word';

test('single occurrence', () => {
    const wrapper = renderer.create(
        <Word isDragging={false}
              occurrences={1}
              occurrence={1}
              word={"hello"}/>
    );
    expect(wrapper).toMatchSnapshot();
});
