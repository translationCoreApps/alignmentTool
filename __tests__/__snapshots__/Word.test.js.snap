// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`single occurrence 1`] = `
<div
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderLeft": "5px solid #44C6FF",
      "boxShadow": "0 1px 4px rgba(0, 0, 0, 0.3), 0 0 40px rgba(0, 0, 0, 0.1) inset",
      "cursor": "move",
      "display": "flex",
      "flexDirection": "row",
      "margin": "10px",
      "opacity": 1,
      "padding": "5px",
    }
  }
>
  <span
    style={
      Object {
        "flex": 1,
      }
    }
  >
    hello
  </span>
</div>
`;
